automatic_data_generation/evaluation/metrics.py

```diff
@@ -154,18 +162,17 @@ def intent_classification(candidates, train_path, input_type):
         This only works for snips dataset for now...
     """
     accuracies = {}
-    # try:
     intent_classifier = train_intent_classifier(train_path, input_type)
     for intent, tokenized_sentences in candidates.items():
         sentences = [' '.join(sentence) for sentence in tokenized_sentences]
         preds = intent_classifier.predict(sentences)
-        accuracy = sum([pred==intent for pred in preds]) / len(preds)
+        corrects = [pred == intent for pred in preds]
+        candidates[intent] = [cand for i, cand in enumerate(candidates[intent])
+                              if corrects[i]]
+        accuracy = sum(corrects) / len(preds)
```

Comment:
I took me a while to understand this, you should explicitly return `candidates` as you modify it and it's used by the other metrics. Best would be to return something like `correctly_classified_candidates` and use this as input for BLEU, originality and transfer.